### PR TITLE
#2544 Kundur VR and PSS in the library, two new preassembled, directo…

### LINKED ISSUE #2544: Add theoretical regulations (AVR, PSS) for Kundur test case to library

### PR DESCRIPTION
**Pull request by ErwanGuichard49**
…ries renamed, references updated (invisible differences)

closes #2544

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+* of simulation tools for power systems.
+*/
+
+model PssKundur "Power system stabilizer based on Kundur's book"
+  import Modelica;
+  import Dynawo.Electrical.SystemBase;
+
+  parameter Types.PerUnit KStab "PSS gain";
+  parameter Types.Time t1 "Phase compensation time constant in s";
+  parameter Types.Time t2 "Phase compensation time constant in s";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:22]: Same name for t1 and t2 ?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  parameter Types.Time tW "Washout time constant in s";
+  parameter Types.VoltageModulePu VsMinPu "Minimum output of PSS in pu (base UNom)";
+  parameter Types.VoltageModulePu VsMaxPu "Maximum output of PSS in pu (base UNom)";
+
+  //Input variables
+  Modelica.Blocks.Interfaces.RealInput omegaPu(start = SystemBase.omega0Pu) "Angular frequency in pu (base omegaNom)" annotation(
+    Placement(visible = true, transformation(origin = {-140, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-46, -30}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput omegaRefPu(start = SystemBase.omega0Pu) "Reference angular frequency in pu (base omegaNom)" annotation(
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:30]: omegaRef0Pu, although it's the same value

Comment (cossartque) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:30]: Is it actually a reference or the grid frequency ?

File: dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt [hunk 3]
```diff
@@ -54,6 +54,8 @@ set(MODELICA_PREASSEMBLED_MODELS
   GeneratorSynchronousThreeWindingsGoverPropVRPropInt.xml
   GeneratorSynchronousFourWindingsTGov1Sexs.xml
   GeneratorSynchronousFourWindingsTGov1SexsPss2A.xml
+  GeneratorSynchronousFourWindingsVRKundur.xml
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt:57]: No governor ?

File: dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt [hunk 4]
```diff
@@ -54,6 +54,8 @@ set(MODELICA_PREASSEMBLED_MODELS
...
+  GeneratorSynchronousFourWindingsVRKundurPssKundur.xml
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt:58]: No governor ?

File: dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundur.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+    Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundur.xml:3]: Copyright to update

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo (hunk 1 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:22]: parameter Types.Time t1 "Phase compensation lead time constant in s";
  parameter Types.Time t2 "Phase compensation lag time constant in s";

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo (hunk 2 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:30]: Agreed and done for omegaRef0Pu.

I do not know, the answer may be in Kundur's book.

File: dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt (hunk 4 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/PreassembledModels/CMakeLists.txt:58]: No

File: dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundur.xml (hunk 5 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundur.xml:3]: Updated

File: dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundurPssKundur.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+    Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/PreassembledModels/GeneratorSynchronousFourWindingsVRKundurPssKundur.xml:3]: Updated

File: examples/DynaSwing/description/DynaSwingExamples.tex [hunk 7]
```diff
@@ -197,15 +197,15 @@ \section{Results}
   \begin{tikzpicture}
     \begin{axis} [ymin = 0, ymax = 3.14, legend style={at={(0.5,-0.1)},anchor=north}]
         \addplot[color=red!50]
-        table[x=time, y=SM_gen_theta]
-        {../Kundur_Example13/reference/outputs_AVR_SetPoint/curves/curves.csv};
+        table[x=time, y=SM_generator_theta]
```
[the remark this answers is not in the document]
Comment (ErwanGuichard49) [reply examples/DynaSwing/description/DynaSwingExamples.tex:200]: Yes, it is compiled successfully.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo (hunk 2 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/PowerSystemStabilizers/Standard/PssKundur.mo:30]: The description is now
_"Grid angular frequency in pu (base omegaNom)"_

**Review by cossartque — APPROVED** [review]

**Pull request merged**